File: third-maximum-number/third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        Arrays.sort(nums);
        int thirdmax=0;
        int changes = 2;
        int currentHighest=nums[nums.length-1];
        for(int i=nums.length-1; i>-1; i--){
            if(nums[i]!=currentHighest && changes>0){
                thirdmax=nums[i];
                currentHighest=nums[i];
                changes--;
            }
            if(changes<=0){
                return thirdmax;
            }
        }
        return nums[nums.length-1];
    }
}

File: 414-third-maximum-number/third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        f = -(2**32)
        s = -(2**32)
        t = -(2**32)
        for val in nums:
            print(f,s,t)
            if val > f:
                temp = f
                f = val
                t = s
                s = temp
            elif val < f and val > s:
                s, t = t, s
                s = val
            elif val < s and val > t:
                t = val
        return t if t > -(2**32) else f



